gulpfile.js

```diff
@@ -787,13 +824,13 @@ function compileJs(srcDir, srcFilename, destDir, options) {
       .pipe(lazywrite())
       .on('end', function() {
         appendToCompiledFile(srcFilename, destDir + '/' + destFilename);
-        logBuildStep('Compiled', srcFilename);
-      }));
+      })).then(() => {
+        endBuildStep('Compiled', srcFilename, startTime);
```

Comment:
I chatted offline with Will, and it appears this is the best kind of measurement we can do given how the parallelism in our build system works.

@erwinmombay, @cramforce, this is now ready for review.